### ISSUE TITLE
2.8.0 DataProxy returned an error

### ISSUE THREAD

**Issue by tedstrauss** [ckan/ckan#4330]
Near duplicate of https://github.com/ckan/ckan/issues/3744
Bug started after upgrading from 2.7 to 2.8.

### CKAN Version if known (or site URL)

2.8

### Please describe the expected behaviour

Data explorer should work with uploaded CSV file.

### Please describe the actual behaviour

> Could not load view: DataProxy returned an error (Data transformation failed. HTTPError: HTTP Error 404: Not Found)

Attaching Apache error log and datapusher error log (below), starting from when I click the 'update' button on add resource page.

```
[Thu Jul 12 11:05:04.643413 2018] [authz_core:debug] [pid 2318] mod_authz_core.c(802): [client 127.0.0.1:42669] AH01626: authorization result of Require all granted: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource_edit/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:04.643465 2018] [authz_core:debug] [pid 2318] mod_authz_core.c(802): [client 127.0.0.1:42669] AH01626: authorization result of <RequireAny>: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource_edit/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:04.643537 2018] [authz_core:debug] [pid 2318] mod_authz_core.c(802): [client 127.0.0.1:42669] AH01626: authorization result of Require all granted: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource_edit/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:04.643553 2018] [authz_core:debug] [pid 2318] mod_authz_core.c(802): [client 127.0.0.1:42669] AH01626: authorization result of <RequireAny>: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource_edit/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:04.846542 2018] [:error] [pid 2313] 2018-07-12 11:05:04,846 DEBUG [ckanext.datapusher.plugin] Submitting resource 402a0806-c609-4ea1-8e5c-ec57caede429 to DataPusher
[Thu Jul 12 11:05:05.131842 2018] [:error] [pid 2313] 2018-07-12 11:05:05,131 CRITI [ckanext.datapusher.plugin] {'status_code': 500, 'message': 'An Error occurred while sending the job: 500 Server Error: Internal Server Error', 'details': u'<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">\\n<html><head>\\n<title>500 Internal Server Error</title>\\n</head><body>\\n<h1>Internal Server Error</h1>\\n<p>The server encountered an internal error or\\nmisconfiguration and was unable to complete\\nyour request.</p>\\n<p>Please contact the server administrator at \\n [no address given] to inform them of the time this error occurred,\\n and the actions you performed just before this error.</p>\\n<p>More information about this error may be available\\nin the server error log.</p>\\n<hr>\\n<address>Apache/2.4.7 (Ubuntu) Server at 0.0.0.0 Port 8800</address>\\n</body></html>\\n'}
[Thu Jul 12 11:05:09.158188 2018] [:error] [pid 2313] 2018-07-12 11:05:09,158 INFO  [ckan.lib.base]  /dataset/the-open-meg-archive/resource_edit/402a0806-c609-4ea1-8e5c-ec57caede429 render time 4.507 seconds
[Thu Jul 12 11:05:09.160850 2018] [headers:debug] [pid 2318] mod_headers.c(845): AH01502: headers: ap_headers_output_filter()
[Thu Jul 12 11:05:09.166096 2018] [authz_core:debug] [pid 2320] mod_authz_core.c(802): [client 127.0.0.1:42674] AH01626: authorization result of Require all granted: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource_edit/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.166121 2018] [authz_core:debug] [pid 2320] mod_authz_core.c(802): [client 127.0.0.1:42674] AH01626: authorization result of <RequireAny>: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource_edit/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.166184 2018] [authz_core:debug] [pid 2320] mod_authz_core.c(802): [client 127.0.0.1:42674] AH01626: authorization result of Require all granted: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource_edit/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.166203 2018] [authz_core:debug] [pid 2320] mod_authz_core.c(802): [client 127.0.0.1:42674] AH01626: authorization result of <RequireAny>: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource_edit/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.191974 2018] [:error] [pid 2314] 2018-07-12 11:05:09,191 WARNI [ckan.lib.maintain] Function _resource_preview() in module ckan.controllers.package has been deprecated and will be removed in a later release of ckan. Resource preview is deprecated. Please use the new resource views
[Thu Jul 12 11:05:09.354584 2018] [:error] [pid 2314] 2018-07-12 11:05:09,354 INFO  [ckan.lib.base]  /dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429 render time 0.184 seconds
[Thu Jul 12 11:05:09.481281 2018] [deflate:debug] [pid 2320] mod_deflate.c(849): [client 127.0.0.1:42674] AH01384: Zlib: Compressed 34219 to 7186 : URL /dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource_edit/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.481377 2018] [headers:debug] [pid 2320] mod_headers.c(845): AH01502: headers: ap_headers_output_filter()
[Thu Jul 12 11:05:09.589039 2018] [authz_core:debug] [pid 2316] mod_authz_core.c(802): [client 127.0.0.1:42676] AH01626: authorization result of Require all granted: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.589087 2018] [authz_core:debug] [pid 2316] mod_authz_core.c(802): [client 127.0.0.1:42676] AH01626: authorization result of <RequireAny>: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.589152 2018] [authz_core:debug] [pid 2316] mod_authz_core.c(802): [client 127.0.0.1:42676] AH01626: authorization result of Require all granted: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.589174 2018] [authz_core:debug] [pid 2316] mod_authz_core.c(802): [client 127.0.0.1:42676] AH01626: authorization result of <RequireAny>: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.633812 2018] [:error] [pid 2313] 2018-07-12 11:05:09,633 INFO  [ckan.lib.base]  /dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429/view/81c668c7-ec26-4c65-b3ab-9cbb9d8b35ba render time 0.040 seconds
[Thu Jul 12 11:05:09.686708 2018] [deflate:debug] [pid 2316] mod_deflate.c(849): [client 127.0.0.1:42676] AH01384: Zlib: Compressed 5885 to 1603 : URL /dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429/view/81c668c7-ec26-4c65-b3ab-9cbb9d8b35ba, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.686845 2018] [headers:debug] [pid 2316] mod_headers.c(845): AH01502: headers: ap_headers_output_filter()
[Thu Jul 12 11:05:09.769698 2018] [authz_core:debug] [pid 2317] mod_authz_core.c(802): [client 127.0.0.1:42679] AH01626: authorization result of Require all granted: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.769763 2018] [authz_core:debug] [pid 2317] mod_authz_core.c(802): [client 127.0.0.1:42679] AH01626: authorization result of <RequireAny>: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.769853 2018] [authz_core:debug] [pid 2317] mod_authz_core.c(802): [client 127.0.0.1:42679] AH01626: authorization result of Require all granted: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.769878 2018] [authz_core:debug] [pid 2317] mod_authz_core.c(802): [client 127.0.0.1:42679] AH01626: authorization result of <RequireAny>: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429
[Thu Jul 12 11:05:09.801694 2018] [headers:debug] [pid 2317] mod_headers.c(845): AH01502: headers: ap_headers_output_filter()
[Thu Jul 12 11:05:09.980252 2018] [authz_core:debug] [pid 2755] mod_authz_core.c(802): [client 127.0.0.1:42680] AH01626: authorization result of Require all granted: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429/view/81c668c7-ec26-4c65-b3ab-9cbb9d8b35ba
[Thu Jul 12 11:05:09.980302 2018] [authz_core:debug] [pid 2755] mod_authz_core.c(802): [client 127.0.0.1:42680] AH01626: authorization result of <RequireAny>: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429/view/81c668c7-ec26-4c65-b3ab-9cbb9d8b35ba
[Thu Jul 12 11:05:09.980528 2018] [authz_core:debug] [pid 2755] mod_authz_core.c(802): [client 127.0.0.1:42680] AH01626: authorization result of Require all granted: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429/view/81c668c7-ec26-4c65-b3ab-9cbb9d8b35ba
[Thu Jul 12 11:05:09.980548 2018] [authz_core:debug] [pid 2755] mod_authz_core.c(802): [client 127.0.0.1:42680] AH01626: authorization result of <RequireAny>: granted, referer: https://open.bic.mni.mcgill.ca/dataset/the-open-meg-archive/resource/402a0806-c609-4ea1-8e5c-ec57caede429/view/81c668c7-ec26-4c65-b3ab-9cbb9d8b35ba
[Thu Jul 12 11:05:10.099680 2018] [headers:debug] [pid 2755] mod_headers.c(845): AH01502: headers: ap_headers_output_filter()
[Thu Jul 12 11:05:10.443344 2018] [authz_core:debug] [pid 2320] mod_authz_core.c(802): [client 127.0.0.1:42681] AH01626: authorization result of Require all granted: granted
[Thu Jul 12 11:05:10.443395 2018] [authz_core:debug] [pid 2320] mod_authz_core.c(802): [client 127.0.0.1:42681] AH01626: authorization result of <RequireAny>: granted
[Thu Jul 12 11:05:10.443464 2018] [authz_core:debug] [pid 2320] mod_authz_core.c(802): [client 127.0.0.1:42681] AH01626: authorization result of Require all granted: granted
[Thu Jul 12 11:05:10.443546 2018] [authz_core:debug] [pid 2320] mod_authz_core.c(802): [client 127.0.0.1:42681] AH01626: authorization result of <RequireAny>: granted
[Thu Jul 12 11:05:10.458517 2018] [:error] [pid 2314] 2018-07-12 11:05:10,458 INFO  [ckan.lib.base]  /dataset/c3a20783-09b3-45d7-9016-94e79b3c6dbc/resource/402a0806-c609-4ea1-8e5c-ec57caede429/download/old_omega-general-screening-questionnaire-omega-general-screening-q-eng-1.csv render time 0.010 seconds
[Thu Jul 12 11:05:10.497055 2018] [:error] [pid 2314] 2018-07-12 11:05:10,496 INFO  [ckan.lib.base]  /dataset/c3a20783-09b3-45d7-9016-94e79b3c6dbc/resource/402a0806-c609-4ea1-8e5c-ec57caede429/download/old_omega-general-screening-questionnaire-omega-general-screening-q-eng-1.csv render time 0.035 seconds
[Thu Jul 12 11:05:10.498891 2018] [headers:debug] [pid 2320] mod_headers.c(845): AH01502: headers: ap_headers_output_filter() 
```

datapusher error log: 

```
[Thu Jul 12 13:15:13.763727 2018] [authz_core:debug] [pid 2755] mod_authz_core.c(802): [client 127.0.0.1:46738] AH01626: authorization result of Require all granted: granted
[Thu Jul 12 13:15:13.763931 2018] [authz_core:debug] [pid 2755] mod_authz_core.c(802): [client 127.0.0.1:46738] AH01626: authorization result of <RequireAny>: granted
[Thu Jul 12 13:15:13.764100 2018] [authz_core:debug] [pid 2755] mod_authz_core.c(802): [client 127.0.0.1:46738] AH01626: authorization result of Require all granted: granted
[Thu Jul 12 13:15:13.764195 2018] [authz_core:debug] [pid 2755] mod_authz_core.c(802): [client 127.0.0.1:46738] AH01626: authorization result of <RequireAny>: granted
[Thu Jul 12 13:15:13.764486 2018] [:info] [pid 2315] [remote 127.0.0.1:37012] mod_wsgi (pid=2315, process='datapusher', application='ckan:8800|'): Loading WSGI script '/etc/ckan/datapusher.wsgi'.
[Thu Jul 12 13:15:13.766471 2018] [:error] [pid 2315] [remote 127.0.0.1:37012] mod_wsgi (pid=2315): Target WSGI script '/etc/ckan/datapusher.wsgi' cannot be loaded as Python module.
[Thu Jul 12 13:15:13.766498 2018] [:error] [pid 2315] [remote 127.0.0.1:37012] mod_wsgi (pid=2315): Exception occurred processing WSGI script '/etc/ckan/datapusher.wsgi'.
[Thu Jul 12 13:15:13.766527 2018] [:error] [pid 2315] [remote 127.0.0.1:37012] Traceback (most recent call last):
[Thu Jul 12 13:15:13.766560 2018] [:error] [pid 2315] [remote 127.0.0.1:37012]   File "/etc/ckan/datapusher.wsgi", line 8, in <module>
[Thu Jul 12 13:15:13.766615 2018] [:error] [pid 2315] [remote 127.0.0.1:37012]     import ckanserviceprovider.web as web
[Thu Jul 12 13:15:13.766637 2018] [:error] [pid 2315] [remote 127.0.0.1:37012]   File "/usr/lib/ckan/datapusher/lib/python2.7/site-packages/ckanserviceprovider/web.py", line 13, in <module>
[Thu Jul 12 13:15:13.766689 2018] [:error] [pid 2315] [remote 127.0.0.1:37012]     import flask.ext.login as flogin
[Thu Jul 12 13:15:13.766720 2018] [:error] [pid 2315] [remote 127.0.0.1:37012] ImportError: No module named ext.login
```

### What steps can be taken to reproduce the issue? 

Upgrade CKAn from 2.7 to 2.8.
Create/edit dataset.
Upload CSV resource.
Try to preview it in Data Explorer.

**Comment by s-chand** [comment]
Yes, this is from the ckanserviceprovider package. Temporary fix is to patch the version of flask to something that works pending when we get a release for 2.8 that comes with the update ckanserviceprovider

**Comment by amercader** [comment]
The latest release of DataPusher (0.0.14) includes the fix for this (ckanserviceprovider 0.0.7). To fix an existing install just install Flask<1.0 (eg Flask==0.12.2) in your DataPusher virtualenv:

     pip install --upgrade Flask<1.0

**Comment by espirado** [comment]
@amercader i am having this similar issue with ubuntu 18 ckan 2.8.4 . There are no logs for datapusher which is odd and i cant seem to find what the issue is since i have followed the manual to the latter. 

**"Bad request - Action name not known: datastore_search"(default) .... when i run the test **

**[pid 95314:tid 139990494263040] 2020-06-24 08:44:44,245 INFO  [ckan.views.api] Action name not known: datastore_search
**
